log4j-core/src/main/java/org/apache/logging/log4j/core/async/DisruptorUtil.java

```diff
@@ -153,4 +155,45 @@ public Long call() {
             throw new IllegalStateException(msg, ex);
         }
     }
+
+    /**
+     * Creates a new {@link Semaphore} or null based on
+     * <pre>AsyncLogger.AsyncQueueFullEnqueueSemaphorePermits</pre>.
+     */
+    static Semaphore createAsyncLoggerQueueFullEnqueueSemaphore() {
+        return createAsyncQueueFullEnqueueSemaphore("AsyncLogger.AsyncQueueFullEnqueueSemaphorePermits");
+    }
+
+    /**
+     * Creates a new {@link Semaphore} or null based on
+     * <pre>AsyncLoggerConfig.AsyncQueueFullEnqueueSemaphorePermits</pre>.
+     */
+    static Semaphore createAsyncConfigQueueFullEnqueueSemaphore() {
+        return createAsyncQueueFullEnqueueSemaphore("AsyncLoggerConfig.AsyncQueueFullEnqueueSemaphorePermits");
+    }
```

Comment:
TODO: Add these properties to our documentation. I'm holding off until we're happy with the design.